src/robusta/core/discovery/discovery.py

```diff
@@ -205,6 +205,14 @@ def is_pod_ready(pod: V1Pod) -> bool:
     return False
 
 
+def is_pod_finished(pod: V1Pod) -> bool:
+    try:
+        # all containers in the pod have terminated, this pod should be removed by GC
+        return pod.status.phase.lower() in ['succeeded', 'failed']
+    except Exception: # phase is an optional field
```

Comment:
Better change to `except AttributeError`, as exactly `AttributeError` is expected.